### PR TITLE
made required changes.

### PR DESCRIPTION
**Pull request by gem-sethiamit**
1> code formatted
2> method name changed
3> wild card removed
-> comments added
-> and some other changes related to exception handling and validation.

### CODE REVIEW

File: src/test/java/com/example/DsStore/service/CustomerServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		Customer customer = new Customer(1, "Amit", "Punjab", 12345);
+		when(customerRepo.save(any(Customer.class))).thenReturn(customer);
+		when(customerRepo.findById(anyInt())).thenReturn(Optional.of(customer));
+
+		customer.setCustomerName("Pulkit");
+		Customer updatedCustomer = customerService.updateCustomer(customer, customer.getCustomerId());
+
+		assertNotNull(updatedCustomer);
+		assertEquals("Pulkit", customer.getCustomerName());
+
```
Comment (abhi-saini-gemini) [inline src/test/java/com/example/DsStore/service/CustomerServiceTest.java:70]: Looks good, but could you check back and see if we can do `verify()` for verifying repository calls made by our service?
If we can/should... Let's do it for all service tests?

Comment (gem-sethiamit) [reply src/test/java/com/example/DsStore/service/CustomerServiceTest.java:70]: yes we can, in some tests i have verified i will do it for all.

File: src/main/java/com/example/DsStore/entities/Product.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	@Size(min = 5, message = "Product Desc must be min of 8 characters")
+	private String productDesc;
+	@NotNull
+	private int price;
+	@JsonFormat(pattern = "dd-MM-yyyy")
+	private Date expiry;
+	@NotNull
+	private int count;
+	@NotNull
+	private boolean Avalibity;
```
Comment (abhi-saini-gemini) [inline src/main/java/com/example/DsStore/entities/Product.java:43]: Would this cause problems during usage or Lombok getter/setter as this is not camelCase?

`private boolean avalibity;`

**Review by abhi-saini-gemini — APPROVED** [review]

File: src/main/java/com/example/DsStore/entities/Product.java (hunk 2 above)
Comment (abhi-saini-gemini) [reply src/main/java/com/example/DsStore/entities/Product.java:43]: Things look good to me.. Let me know if above question is concern or not.

**Pull request merged**